Tests/UtilityBeltDataTests/Protocols/CoreDataOperatorTesting.swift

```diff
@@ -7,68 +7,284 @@ import XCTest
 
 protocol CoreDataOperatorTesting: XCTestCase {
     var coreDataOperator: CoreDataOperator { get set }
+    
+    func testNewInstance()
+    
+    func testExists()
+    func testExistsWithPredicate()
+    
+    func testCount()
+    func testCountWithPredicate()
+    
+    func testDeleteSingleObject()
+    func testDeleteAllObjects()
+    func testDeleteAllObjectsWithPredicate()
 }
 
 extension CoreDataOperatorTesting {
-    func count(file: StaticString = #file, line: UInt = #line) {
-        self.measureAndCatch {
-            let userCount = try self.coreDataOperator.count(of: User.self)
+    // MARK: Create
+    
+    func verifyNewInstanceSucceeds(file: StaticString = #file, line: UInt = #line) {
+        do {
+            // Create a new user instance
+            let user = try XCTUnwrap(self.coreDataOperator.newInstance(of: User.self),
+                                     file: file,
+                                     line: line)
             
-            XCTAssertEqual(userCount, 4, file: file, line: line)
+            // Verify the entity name is correct
+            XCTAssertEqual(user.entity.name,
+                           "User",
+                           file: file,
+                           line: line)
+        } catch {
+            XCTFail(error.localizedDescription, file: file, line: line)
         }
     }
     
-    func delete(file: StaticString = #file, line: UInt = #line) {
-        self.measureAndCatch {
-            try self.coreDataOperator.deleteAll(of: User.self)
-            let userExists = try self.coreDataOperator.exists(User.self)
+    func verifyExistsSucceeds(file: StaticString = #file, line: UInt = #line) {
```

Comment:
I'd be curious how this function works without the `do/catch`, given that it's not actually an `XCTest` function that's throwing. `do/catch` with our own `try` functions might need to stay. Can you test this out locally? Curious for my own sake too! 

I'm willing to bet that the assumption you made here is 100% correct, but wondering if `XCTest` will surprise us by figuring out the `file` and `line` and how it's associated with the `try` functions passed into the assertions.